Fix bug where precision is not calculated correctly with whitespace in field

al-niessner opened the pull request:
## 🗒️ Summary
Added check that if delimited table has trailing spaces on number to flag it as data type mismatch error instead of precision.

## ⚙️ Test Data and/or Report
See automated unit tests below (should pass)

## ♻️ Related Issues
Closes #681

Commit:
add unit testing first

Commit:
fix typo

jordanpadams reviewed (changes requested):
@al-niessner one thing we can't forget here is in a delimited table, everything within the delimiters is considered part of the "field value", making spaces within any numeric field invalid.

This only applies to delimited tables. You can pad in character tables since they are fixed width fields.

From the Standards Reference:
<img width="643" alt="Screenshot 2023-10-04 at 10 36 55 AM" src="https://github.com/NASA-PDS/validate/assets/33492486/84454264-6ec6-4ff5-a7f8-5e6748209849">

al-niessner committed:
Update FieldValueValidator.java

nutjob4life reviewed (approved):
LGTM 👍

jordanpadams commented on src/main/java/gov/nasa/pds/tools/validate/content/table/FieldValueValidator.java:
```diff
@@ -840,13 +840,20 @@ private void checkFormat(String value, String format, int fieldIndex,
-                    + ", got " + length + ").",
-                recordLocation, fieldIndex);
             isValid = false;
+            if (value.trim().length() == value.length()) {
```
@al-niessner This fixes the whitespace padding when a `field_format` or `validation_format` are specified, but it looks like this may overall be a more inherent bug in handling of whitespace in delimited tables. When I remove the `field_format` attributes from the `ff_del.xml` label, it passes, but it should fail. I created a new requirement here that ties to this handling of whitespace in tables more explicitly: https://github.com/NASA-PDS/validate/issues/722

I would just merge and then work on that requirement, but I think this solution will immediately be changed based upon that requirement

al-niessner replied:
done

jordanpadams reviewed (changes requested):
See review comment

al-niessner committed:
Update FieldValueValidator.java

make checking for leading/trailing spaces in delimited table default data type error

jordanpadams committed:
Revert "Update FieldValueValidator.java"

This reverts commit 026b850faaeb38bd020f5a4b6cb161177a5edce5.

jordanpadams committed:
Remove check for whitespace

jordanpadams reviewed (approved):
🎉
